packages/api/src/routes/ehr/elation/appointment-webhook.ts

```diff
@@ -1,23 +1,62 @@
+import { buildEhrSyncPatientHandler } from "@metriport/core/external/ehr/sync-patient/ehr-sync-patient-factory";
+import { ElationSecondaryMappings } from "@metriport/shared/interface/external/ehr/elation/cx-mapping";
+import { elationAppointmentEventSchema } from "@metriport/shared/interface/external/ehr/elation/event";
+import { EhrSources } from "@metriport/shared/interface/external/ehr/source";
 import { Request, Response } from "express";
 import Router from "express-promise-router";
 import httpStatus from "http-status";
+import { getCxMappingOrFail } from "../../../command/mapping/cx";
+import { processAsyncError } from "../../../errors";
+import { updateOrCreateElationPatientMetadata } from "../../../external/ehr/elation/command/sync-patient";
 import { handleParams } from "../../helpers/handle-params";
 import { requestLogger } from "../../helpers/request-logger";
-import { asyncHandler } from "../../util";
+import { asyncHandler, getCxIdOrFail, getFromQueryOrFail } from "../../util";
 
 const router = Router();
 
 /**
  * POST /ehr/webhook/elation/appointments
  *
- * Currently does nothing.
+ * Tries to retrieve the matching Metriport patient on appointment created
+ * @returns HTTP 200 OK on successful processing.
  */
 router.post(
   "/",
   handleParams,
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
-    return res.status(httpStatus.NOT_IMPLEMENTED).send();
+    const cxId = getCxIdOrFail(req);
+    const elationPracticeId = getFromQueryOrFail("practiceId", req);
+    const event = elationAppointmentEventSchema.parse(req.body);
+    if (event.action === "deleted") {
+      console.log(`Appointment event is a deleted event for appointment ${event.data.id}`);
+      return res.sendStatus(httpStatus.OK);
+    }
+    if (event.data.created_date !== event.data.last_modified_date) {
+      console.log(`Appointment event is not a created event for appointment ${event.data.id}`);
+      return res.sendStatus(httpStatus.OK);
+    }
+    const cxMapping = await getCxMappingOrFail({
+      externalId: elationPracticeId,
+      source: EhrSources.elation,
+    });
+    const secondaryMappings = cxMapping.secondaryMappings as ElationSecondaryMappings;
```

Comment:
ditto: specific function per EHR, and fine to do that on a follow-up PR given the deadline